src/data_structure/Board.java

```diff
@@ -84,32 +83,20 @@ private void fillGameBoard(Field2D[][] initialBoard) {
         }
     }
 
-    private boolean findNewFilled() {
-        boolean found = false;
-        for (FieldInfo column : columnsInfo) {
-            for (FieldWritable writable : column.getFields()) {
-                if (writable.getState() == FieldWritable.State.UNFILLED && writable.getPossibilitiesCount() == 1) {
-                    writable.setState(FieldWritable.State.FILLED);
-                    found = true;
-                }
-            }
-        }
-        return found;
-    }
-
     private void fillValuesFromShortcut(String shortcut) {
         int indexInShortcut = 0;
 
         for (int i = 0; i < height; ++i) {
             for (int j = 0; j < width; ++j) {
                 if (gameBoard[j][i].getType() == Field2D.Type.WRITABLE) {
                     if (shortcut.charAt(indexInShortcut) != ASCIZEROCODE)
-                        this.setField(j, i, (int) shortcut.charAt(indexInShortcut) - ASCIZEROCODE);
+                        gameBoard[j][i].getWritable().setValue((int) shortcut.charAt(indexInShortcut) - ASCIZEROCODE);
                     ++indexInShortcut;
                 }
             }
         }
-        //setPossibilities(); // TODO: is this needed or not?
+        setPossibilitiesBasedOnTemplate();
```

Comment:
Is this needed? Łukasz said on previous PR, that we don't need setting possibilities here

Reply:
i changed line 93 so we need it